js/components/generation/select-sources-popup.js

```diff
@@ -24,7 +24,7 @@ define([
 
 			this.showModal = params.showModal;
 			this.selectedSources = params.selectedSources;
-			this.sourceOptions = ko.computed(() => params.sources().map(s => ({ source: s, selected: ko.computed(() => this.selectedSources().includes(s)) })));
+			this.sourceOptions = ko.computed(() => params.sources().map(s => ({ source: s, selected: ko.computed(() => this.selectedSources().find(ss => ss.sourceKey === s.sourceKey)) })));
```

Comment:
Why this was switched to sourceKey?

Reply:
Well, because source object is changing during polling